### PR TITLE
[Oreo] Emoji Sample

### PR DESCRIPTION
**Pull request by gonzalonm**
Ported sample from https://github.com/googlesamples/android-EmojiCompat/

### CODE REVIEW

File: android-o/EmojiCompat/README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ Using this sample requires the Android 8.0 (API 26) and the Xamarin.Android 7.5
 Screenshots
 -------------
 
-<img src="screenshots/Main.png" height="400" alt="Screenshot"/> 
+<img src=“Screenshots/Main.png" height="400" alt="Screenshot"/>
```
Comment (conceptdev) [inline android-o/EmojiCompat/README.md]: markdown image format would be as follows (providing good alt text is essential for accessibility and SEO). don't rely on html image tag and `height` to scale the image, instead upload an image of the correct size 

```
![Description of image for accessibility](Screenshots/Main.png)

```

Comment (gonzalonm) [inline android-o/EmojiCompat/README.md]: Actually, that section (Screenshot) was copied from original README and I removed it. It is not in current README file. If a screenshot link is needed, I can add it as you mentioned.

File: android-o/EmojiCompat/EmojiCompat/CustomTextView.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		EmojiTextViewHelper mEmojiTextViewHelper;
+
+		public CustomTextView(IntPtr handle, JniHandleOwnership transfer) : base(handle, transfer) { }
+
+		public CustomTextView(Context context): this(context, null) {}
+
+		public CustomTextView(Context context, IAttributeSet attrs) : this(context, attrs, 0) {}
+
+		public CustomTextView(Context context, IAttributeSet attrs, int defStyleAttr) 
+			: base(context, attrs, defStyleAttr) {
```
[the remark this answers is not in the document]
Comment (gonzalonm) [reply android-o/EmojiCompat/EmojiCompat/CustomTextView.cs]: Done

File: android-o/EmojiCompat/EmojiCompat/MainActivity.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+			// Regular TextView without EmojiCompat support; you have to manually process the text
+			var regularTextView = (TextView) FindViewById(Resource.Id.regular_text_view);
+			Get().RegisterInitCallback(new InitCallbackImpl {Context = this, RegularTextView = regularTextView});
+
+			// Custom TextView
+			var customTextView = (TextView) FindViewById(Resource.Id.emoji_custom_text_view);
+			customTextView.Text = GetString(Resource.String.custom_text_view, Emoji);
+		}
+
+		class InitCallbackImpl : InitCallback {
```
[the remark this answers is not in the document]
Comment (gonzalonm) [reply android-o/EmojiCompat/EmojiCompat/MainActivity.cs]: Done

**Comment by gonzalonm** [comment]
@pjcollins I've applied your feedback. The regularTextView is not displayed because the following line returns null:

`EmojiCompat.Get().Process(Text)`

I'm not sure why it is returning null. I set the same configuration like original version. It could be a bug, will review with @Redth

**Comment by gonzalonm** [comment]
Debugging it I noticed that the string to set is built fine before call to `Get().Process(Text)`.

screen shot 2017-08-29 at 8 56 08 am

**Comment by gonzalonm** [comment]
Using `RegularTextView.TextFormatted = Get().ProcessFormatted(new Java.Lang.String(Text));` for the moment and filed bug https://bugzilla.xamarin.com/show_bug.cgi?id=59472

**Comment by pjcollins** [comment]
@gonzalonm Actually on second thought, let's update the sample's Namespace to avoid namespace conflicts between this project and `Android.Support.Text.Emoji.EmojiCompat`. Perhaps we can just update to `EmojiCompatSample`?

**Comment by gonzalonm** [comment]
@pjcollins namespace was renamed to `EmojiCompatSample` and using `EmojiCompat.*` properly without `using static` sentence.

**Pull request merged**